### PR TITLE
character-repertoire issues

### PR DESCRIPTION
**Pull request by timbray**

### CODE REVIEW

File: draft-ietf-jsonpath-base.md
Comment (cabo) [inline draft-ietf-jsonpath-base.md:529]:
```suggestion
Unicode scalar values as described
```

File: draft-ietf-jsonpath-base.md
Comment (cabo) [inline draft-ietf-jsonpath-base.md:834]: No, that is not the way we have defined JSONPath to be.
(Even if we wanted to change this now, \u1F609 is ambiguous (U+1f60 + character 9?), so we'd need to adopt one of the syntaxes \U or \u{}...)

**Review by cabo — CHANGES_REQUESTED** [review]
Looks good, except for the attempt to get rid of the ugly JSON escapes.

**Review by cabo — APPROVED** [review]
(Ignoring the change to lib) these changes now look good.  They do show the tedium of skipping the surrogate code points in each ABNF range covering that space, but that may be the best minimum change we can do here.

**Comment by cabo** [comment]
> What's the outcome of this change? 

Over on the jsonpath mailing list, I wrote:

> > The recent extensive discussions on [art@ietf.org](mailto:art@ietf.org) and [i18n@ietf.org](mailto:i18n@ietf.org) have heightened our senses for potential misunderstandings of the way JSONPath handles Unicode.  Pull request 512 [1] is intended to address this.
> >
> > Note that while this PR occurs during IESG processing, this is not based on an IESG comment.
Some of us still would like to make this change before the document is approved.
> >
> > These changes are editorial in nature in that the technical content does not change, it is just expressed in a way that may give less opportunities for confusion.  
> >
> > So I’m requesting that the WG please have a look and speak up if you see a problem with this editorial improvement.

I think that should answer your question.

> Is it that surrogate pairs are no longer to be supported? 

There is no technical change here; JSONPath parsers never see surrogate pairs; these happen only inside UTF-16.
The ABNF changes make it more obvious that lone surrogate code points cannot occur in a JSONPath query; that is already the case now (see Section 1.1 in conjunction with RFC 3269).

> What's the reasoning?
> 
> Providing some context in the description on PRs would be good.

Sorry for putting this in the WG message only; the background is some extensive discussion on art@ and i18n@ that we don't want to repeat here in the JSONPath WG.

**Comment by gregsdennis** [comment]
> I think that should answer your question.

No, that doesn't answer my question.  I saw that email and still asked.

> the background is some extensive discussion on art@ and i18n@ that we don't want to repeat here in the JSONPath WG.

Because I'm not subscribed to those, is there a summary?

**Comment by timbray** [comment]
Basically, the JSONPath draft has always said that the character
repertoire for JSONPath is “Unicode scalar values”, which is all the code
points except surrogates, which only have meaning in UTF-16.  The recent
changes did a few s/code point/scalar value/ and changed one piece of ABNF
to make it clear that surrogates are not supported.

**Comment by gregsdennis** [comment]
Thank you

**Review by glyn — APPROVED** [review]
LGTM, once the lib changes are reverted.

File: sourcecode/abnf/jsonpath_collected_abnf
Comment (cabo) [inline sourcecode/abnf/jsonpath_collected_abnf:137]: Oops.  This should be %x80, not %x7E (~ is not a good name character).

File: draft-ietf-jsonpath-base.md
Comment (cabo) [inline draft-ietf-jsonpath-base.md:1815]: (See above0

**Review by glyn — APPROVED** [review]
@cabo Thanks for fixing the typo! I was just about to dig into it...

**Pull request merged**
